File: archetype-scala/src/main/resources/archetype-resources/src/main/scala/view/HelloController.scala
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.view

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import collection.JavaConverters._

@Controller
class HelloController {

  @RequestMapping(Array("/hello"))
  def hello() = {
    println("HelloController: Passing through...")
    Map("hello" -> "hello controller") asJava
  }
}

File: example/src/main/java/thirtytwo/degrees/halfpipe/example/view/HelloController.java
package thirtytwo.degrees.halfpipe.example.view;

import com.yammer.metrics.annotation.Timed;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import thirtytwo.degrees.halfpipe.logging.Log;

import java.io.IOException;
import java.io.Writer;

@Controller
public class HelloController {
    private static final Log LOG = Log.forThisClass();

    @RequestMapping(value = "/hello")
    @Timed
    public void home(Writer out) throws IOException {
        LOG.warn("HelloController: Passing through...");
        out.write("hello controller");
        out.flush();
        out.close();
    }
}
